### ISSUE TITLE
[bug] Can't run alias task with custom cli arguments

### ISSUE THREAD

**Issue by felicienfrancois** [shootaroo/jit-grunt#36]
my grunt file:
```javascript
module.exports = function (grunt) {

    // get my custom cli argument (http://gruntjs.com/api/grunt.option)
    var target = grunt.option('target');

    require("jit-grunt")(grunt);

    grunt.initConfig({
         ...
    });

    // My alias task
    grunt.registerTask("build", ["clean:build", "copy:build", "uglify:build"]);
}
```

Without jit-grunt, the following command execute the alias task `build` with `target` option set to `test`
```bash
grunt build --target=test
```

With jit-grunt I get the following error:
```bash
Loading "GruntFile.js" tasks...ERROR
>> TypeError: undefined is not a function

jit-grunt: Plugin for the "build" task not found.
If you have installed the plugin already, please setting the static mapping.
See https://github.com/shootaroo/jit-grunt#static-mappings

Warning: Task "build" failed. Use --force to continue.

Aborted due to warnings.
```

**Comment by shootaroo** [comment]
Add the `--stack` option, please tell me the stack trace.
```bash
grunt build --target test --stack 
```

**Comment by felicienfrancois** [comment]
The stack:
```
Error: Task "build" failed.
    at Object.task.fn (...\node_modules\jit-grunt\lib\jit-grunt.js:114:20)
    at Object.<anonymous> (...\node_modules\grunt\lib\util\task.js:301:30)
    at Task.runTaskFn (...\node_modules\grunt\lib\util\task.js:251:24)
    at Task.<anonymous> (...\node_modules\grunt\lib\util\task.js:300:12)
    at Task.start (...\node_modules\grunt\lib\util\task.js:309:5)
    at Object.grunt.tasks (...\node_modules\grunt\lib\grunt.js:164:8)
    at Object.module.exports [as cli] (...\node_modules\grunt\lib\grunt\cli.js:38:9)
    at Object.<anonymous> (...\AppData\Roaming\npm\node_modules\grunt-cli\bin\grunt:45:20)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
```

**Comment by felicienfrancois** [comment]
In fact, the mistake was my side.
I made a mistake (typo) in my Gruntfile that was evaluated only when a target option was provided.

The error (`>> TypeError: undefined is not a function`) was triggered after `require("jit-grunt")(grunt);` but before registering the alias task so it was not declared.

Then, my fault was I had not used exacly the same Gruntfile when testing without jit-grunt (I commented some parts to avoid the need to register all plugins manually) and so I did not get the error.
